sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/PlanService.java

```diff
@@ -1,14 +1,52 @@
 package com.kyriba.conference.sponsorship.service;
 
+import com.kyriba.conference.sponsorship.dao.PlanRepository;
+import com.kyriba.conference.sponsorship.dao.SponsorRepository;
+import com.kyriba.conference.sponsorship.domain.Plan;
+import com.kyriba.conference.sponsorship.domain.PlanCategory;
+import com.kyriba.conference.sponsorship.domain.Sponsor;
+import com.kyriba.conference.sponsorship.domain.dto.PlanDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.transaction.Transactional;
 
 
 /**
  * @author M-ASL
  * @since v1.0
  */
 @Service
+@Transactional
+@RequiredArgsConstructor
 public class PlanService
 {
-  // todo-asl
+  private final PlanRepository planRepository;
+  private final SponsorRepository sponsorRepository;
+
+
+  public PlanDto readPlan(Long id)
+  {
+    return planRepository.findById(id).map(Plan::toDto)
+        .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Plan Not Found"));

```

Comment:
No HTTP knowledge at business logic level, throw business exceptions and then transform them in controller to desired error code. Alternatively you could return Optional.

Reply:
Can we throw org.springframework.data.rest.webmvc.ResourceNotFoundException or have to create something similar?

Reply:
Ideally this is your own business exception (then it is unlear for calling side that issue may happen) or Optional (then it is hard to miss this fact).